src/go/rpk/pkg/config/weak_test.go

```diff
@@ -0,0 +1,555 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestWeakBool(t *testing.T) {
+	type testWeakBool struct {
+		Wb weakBool `yaml:"wb"`
+	}
+	for _, test := range []struct {
+		name   string
+		data   string
+		exp    bool
+		expErr bool
+	}{
+		{
+			name: "boolean:true",
+			data: "wb: true",
+			exp:  true,
+		},
+		{
+			name: "boolean:false",
+			data: "wb: false",
+			exp:  false,
+		},
+		{
+			name: "int:0",
+			data: "wb: 0",
+			exp:  false,
+		},
+		{
+			name: "non-zero int",
+			data: "wb: 12",
+			exp:  true,
+		},
+		{
+			name: "string:true",
+			data: `wb: "true"`,
+			exp:  true,
+		},
+		{
+			name: "string:false",
+			data: `wb: "false"`,
+			exp:  false,
+		},
+		{
+			name:   "error with unsupported string",
+			data:   `wb: "falsity"`,
+			expErr: true,
+		},
+		{
+			name:   "error with float type",
+			data:   `wb: 123.123`,
+			expErr: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			ts := testWeakBool{}
```

Comment:
How about having this be
```go
var ts struct { Wb weakBool `yaml:"wb"` }
```
to get rid of the type way at the top of the function?